### ISSUE TITLE
ReferenceError: __name is not defined (LESS stylesheets)

### ISSUE THREAD

**Issue by kaiyoma** [esbuild-kit/tsx#174]
### Bug description

I'm trying to migrate my webpack app from Babel to `tsx` and I'm running into a problem. The webpack build mostly completes, but then spits out hundreds of errors, one for every LESS stylesheet in the project. Each error looks like this:

```
ERROR in ./src/modules/topology/components/ClusteredLinkInfo/styles.less (./node_modules/.pnpm/css-loader@5.1.3_webpack@4.44.2/node_modules/css-loader/dist/cjs.js!./node_modules/.pnpm/postcss-loader@4.2.0_cevnyqd5mhktybfrpstjugctw4/node_modules/postcss-loader/dist/cjs.js??ref--15-2!./node_modules/.pnpm/less-loader@7.2.1_less@4.1.3+webpack@4.44.2/node_modules/less-loader/dist/cjs.js??ref--15-3!./src/modules/topology/components/ClusteredLinkInfo/styles.less)
Module build failed (from ./node_modules/.pnpm/postcss-loader@4.2.0_cevnyqd5mhktybfrpstjugctw4/node_modules/postcss-loader/dist/cjs.js):
ReferenceError: __name is not defined
    at SortTemplate (eval at cloneSort (C:\Users\kgetz\Work\event-viewer\node_modules\.pnpm\source-map-js@1.0.2\node_modules\source-map-js\lib\quick-sort.js:2:861), <anonymous>:3:107)
    at cloneSort (c:\Users\kgetz\Work\event-viewer\node_modules\.pnpm\source-map-js@1.0.2\node_modules\source-map-js\lib\quick-sort.js:112:10)
    at exports.quickSort (c:\Users\kgetz\Work\event-viewer\node_modules\.pnpm\source-map-js@1.0.2\node_modules\source-map-js\lib\quick-sort.js:128:19)
    at BasicSourceMapConsumer.SourceMapConsumer_parseMappings (c:\Users\kgetz\Work\event-viewer\node_modules\.pnpm\source-map-js@1.0.2\node_modules\source-map-js\lib\source-map-consumer.js:604:9)
    at BasicSourceMapConsumer.get (c:\Users\kgetz\Work\event-viewer\node_modules\.pnpm\source-map-js@1.0.2\node_modules\source-map-js\lib\source-map-consumer.js:70:12)
    at BasicSourceMapConsumer.SourceMapConsumer_originalPositionFor (c:\Users\kgetz\Work\event-viewer\node_modules\.pnpm\source-map-js@1.0.2\node_modules\source-map-js\lib\source-map-consumer.js:694:12)
    at SourceMapGenerator.<anonymous> (c:\Users\kgetz\Work\event-viewer\node_modules\.pnpm\source-map-js@1.0.2\node_modules\source-map-js\lib\source-map-generator.js:207:43)
    at Array.forEach (<anonymous>)
    at MappingList.MappingList_forEach (c:\Users\kgetz\Work\event-viewer\node_modules\.pnpm\source-map-js@1.0.2\node_modules\source-map-js\lib\mapping-list.js:44:17)
    at SourceMapGenerator.SourceMapGenerator_applySourceMap (c:\Users\kgetz\Work\event-viewer\node_modules\.pnpm\source-map-js@1.0.2\node_modules\source-map-js\lib\source-map-generator.js:204:20)
 @ ./src/modules/topology/components/ClusteredLinkInfo/styles.less 2:12-411 9:17-24 13:7-14 15:20-27 17:4-30:5 19:6-29:7 20:38-45 26:26-33 28:21-28 38:15-22
 @ ./src/modules/topology/components/ClusteredLinkInfo/index.jsx
 @ ./src/modules/topology/containers/Sidebar/index.jsx
 @ ./src/modules/topology/containers/App/index.jsx
 @ ./src/modules/topology/index.js
 @ ./src/containers/Main/constants.ts
 @ ./src/containers/Main/index.tsx
 @ ./index.tsx
 @ multi webpack-dev-server/client?http://localhost:3001 webpack/hot/only-dev-server ./index
```

### Reproduction

This is a very large enterprise project, so I can't provide a repro easily right now. I'm happy to work with you and provide more information, because I'd really love to ditch Babel.

### Environment

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz
    Memory: 5.40 GB / 15.71 GB
  Binaries:
    Node: 16.19.0 - C:\Program Files\Node.js\node.EXE
    npm: 9.2.0 - C:\Program Files\Node.js\npm.CMD
  npmPackages:
    tsx: 3.12.2 => 3.12.2
```

### Can you work on a fix?

- [ ] I’m interested in opening a pull request to address this issue.

**Comment by privatenumber** [comment]
Duplicate of https://github.com/esbuild-kit/tsx/issues/113 https://github.com/esbuild-kit/tsx/issues/87

You can debug your project to identify which dependency is using `eval` or Workers and see if it offers an option to disable it. For example, in `terser-webpack-plugin`: https://github.com/esbuild-kit/tsx/issues/87#issuecomment-1226117760

**Comment by kaiyoma** [comment]
Hmm, I tried setting `parallel: false` and it didn't change anything. How do I determine which webpack dependencies are at fault? is there a list somewhere of webpack plugins that are known to be incompatible with tsx?

**Comment by privatenumber** [comment]
> Hmm, I tried setting parallel: false and it didn't change anything.

That was just an example for `terser-webpack-plugin`.

> How do I determine which webpack dependencies are at fault?

It's in your error stack:
```
    at SortTemplate (eval at cloneSort (C:\Users\kgetz\Work\event-viewer\node_modules\.pnpm\source-map-js@1.0.2\node_modules\source-map-js\lib\quick-sort.js:2:861), <anonymous>:3:107)
```

> Is there a list somewhere of webpack plugins that are known to be incompatible with tsx?

No. As the issues I referenced indicate, tsx doesn't support `eval`. So any code that uses `eval` is incompatible with tsx. Not sure if it makes sense to make a list for that.


Why are you using tsx to run Webpack, anyway?

**Comment by kaiyoma** [comment]
We're currently using Node + Babel to run webpack, but in general we're trying to move off of Babel. We've migrated all our other tooling and packages to use esbuild and webpack is the last holdout. Eventually I want to move to something like Vite, but we need to use webpack for now. esbuild was a great drop-in replacement for rollup, so I wanted to see if it would work for webpack too.

**Comment by privatenumber** [comment]
> We're currently using Node + Babel to run webpack

Why?

**Comment by kaiyoma** [comment]
Sorry, I should clarify that by "webpack" I mean the webpack dev server. We have a server written in TypeScript that needs to be run.

**Comment by privatenumber** [comment]
That makes a lot more sense.

In the future, tsx may become smarter about excluding dependencies from transpilation when unnecessary. That may unblock your issue.

**Comment by fregante** [comment]
The repro is super basic, tsx just cannot be used as a loader with webpack:

```sh
# Package.json
npm init -y
npm install webpack webpack-cli cross-env tsx 

# Files
echo 'module.exports = {}' > webpack.config.ts
echo 'console.log(1)' > index.js

# Command
npx cross-env NODE_OPTIONS="--loader tsx" webpack ./index.js
```

This results into

```
(node:89716) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:89716) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
assets by status 99 bytes [cached] 1 asset
./index.js 15 bytes [built] [code generated]

ERROR in main.js
main.js from Terser plugin
__name is not defined
ReferenceError: __name is not defined
    at terserMinify (eval at transform (./node_modules/terser-webpack-plugin/dist/minify.js:2:448), <anonymous>:4:326)
    at minify (./node_modules/terser-webpack-plugin/dist/minify.js:23:10)
    at Object.transform (./node_modules/terser-webpack-plugin/dist/minify.js:44:10)
    at execFunction (./node_modules/jest-worker/build/workers/threadChild.js:148:17)
    at execHelper (./node_modules/jest-worker/build/workers/threadChild.js:127:5)
    at execMethod (./node_modules/jest-worker/build/workers/threadChild.js:131:5)
    at MessagePort.messageListener (./node_modules/jest-worker/build/workers/threadChild.js:49:7)
    at [nodejs.internal.kHybridDispatch] (node:internal/event_target:735:20)
    at exports.emitMessage (node:internal/per_context/messageport:23:28)

webpack 5.74.0 compiled with 1 error and 1 warning in 217 ms
```

Tested on Node 18

**Comment by RichieChoo** [comment]
Reproduction!!

If used tsx to run node.js cli app which has webpack , **tsx** will add **__name** for webpack hot module replacement function statement
